test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -665,7 +665,7 @@ public static void assertVersionSerializable(Version version, Streamable streama
             BytesReference orig;
             try {
                 orig = serialize(version, streamable);
-            } catch (IllegalArgumentException e) {
+            } catch (IllegalArgumentException | IllegalStateException e) {
```

Comment:
Use IllegalArgumentException for checkVersion functionality as well. It fits better than IllegalStateException I think (the node does not need to look at current state to reject serializing the message but just at the arguments).